core/consensus/babe/src/lib.rs

```diff
@@ -717,8 +718,8 @@ fn median_algorithm(
 			t + Duration::new(secs, nanos)
 		}).collect();
 
-		// FIXME #2926: use a selection algorithm instead of a full sorting algorithm.
-		new_list.sort_unstable();
+		// Use a partial sort to move the median timestamp to the middle of the list
+		select(&mut new_list, num_timestamps / 2);
```

Comment:
```suggestion
		pdqselect::select(&mut new_list, num_timestamps / 2);
```

Reply:
nitpick: maybe easier to understand what this is just by reading this line alone. feel free to ignore it though :)

Reply:
yes, i find this style much better for free functions - it means that whenever you see a function call without a module prefix, that it's defined in the same file as you're reading right now.